extension.js

```diff
@@ -36,34 +36,40 @@ exports.activate = function activate(context) {
 function createSnippet(snippetFilename, editor) {
     fs.readFile(snippetFilename, 'utf8', function(e, data) {
         console.log(data);
-        var jsonData = json.parse(data, null, false);
-        jsonData = addSelectedSnippet(jsonData, editor);
-        updateSnippetFile(snippetFilename, jsonData);
+        var jsonData = (data ? json.parse(data, null, false) : {});
```

Comment:
my biggest concern here is that we would be deleting all of people's snippets if something goes wrong, so I'd prefer this to be an explicit action by the user (give them the option to either edit the file or to recreate it, or maybe even consider backing it up.)